salt/state.py

```diff
@@ -440,75 +475,76 @@ def _verify_high(high: dict) -> list[str]:
-                                    req_key, req_val = next(iter(req.items()))
-                                    if "." in req_key:
-                                        errors.append(
-                                            f"Invalid requisite type '{req_key}' "
-                                            f"in state '{id_}', in SLS "
-                                            f"'{ body['__sls__']}'. Requisite types must "
-                                            "not contain dots, did you "
-                                            f"mean '{req_key[: req_key.find('.')]}'?"
-                                        )
-                                    if not ishashable(req_val):
-                                        errors.append(
-                                            f'Illegal requisite "{req_val}" '
-                                            f'in SLS "{body["__sls__"]}", '
-                                            "please check your syntax.\n"
-                                        )
-                                        continue
-                            # Make sure that there is only one key in the
-                            # dict
-                            if len(list(arg)) != 1:
-                                errors.append(
-                                    "Multiple dictionaries defined in "
-                                    f"argument of state '{id_}' in SLS '{body['__sls__']}'"
-                                )
-                if not fun_count:
+                        first_key, first_value = next(iter(arg.items()))
+                        _verify_high_state_key_value_arg(
+                            id_, first_key, first_value, body, errors
+                        )
+            if not fun_count:
+                errors.append(
+                    f"No function declared in state '{id_}' in SLS "
+                    f"'{body['__sls__']}'"
+                )
+            elif fun_count > 1:
+                funs = [state.split(".", maxsplit=1)[1]] if "." in state else []
+                funs.extend(arg for arg in body[state] if isinstance(arg, str))
+                errors.append(
+                    f"Too many functions declared in state '{id_}' in "
+                    f"SLS '{body['__sls__']}'. Please choose one of "
+                    "the following: " + ", ".join(funs)
+                )
+    return errors
+
+
+def _verify_high_state_key_value_arg(
+    id_: str, arg_name: str, value, body: dict, errors: list[str]
+):
+    if arg_name == "names":
+        if not isinstance(value, list):
+            errors.append(
+                "The 'names' argument in state "
+                f"'{id_}' in SLS '{body['__sls__']}' needs to be "
+                "formed as a list"
+            )
+    elif arg_name in STATE_REQUISITE_KEYWORDS:
+        # if the arg is require or
+        # watch, it must be a list.
```

Comment:
```suggestion
        # if the arg is a requisite,
        # it must be a list.
```

nit: I don't think it's specific to `require` and `watch`.